source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -76,65 +143,77 @@ Http::FilterHeadersStatus CacheFilter::encodeHeaders(Http::ResponseHeaderMap& he
 }
 
 Http::FilterDataStatus CacheFilter::encodeData(Buffer::Instance& data, bool end_stream) {
+  if (skip_encoding_) {
+    return Http::FilterDataStatus::Continue;
+  }
   if (insert_) {
-    ENVOY_STREAM_LOG(debug, "CacheFilter::encodeHeaders inserting body", *encoder_callbacks_);
+    ENVOY_STREAM_LOG(debug, "CacheFilter::encodeData inserting body", *encoder_callbacks_);
     // TODO(toddmgreer): Wait for the cache if necessary.
     insert_->insertBody(
         data, [](bool) {}, end_stream);
```

Comment:
What do you think about adding a `return Http::FilterDataStatus::Continue;` here to make it more explicit? I feel like the ideal is either none of a function's branches return, or all of them do.